libs/geometry/src/TangentSpaceMeshInternal.h

```diff
@@ -22,11 +22,42 @@
 #include <math/mat3.h>
 #include <math/norm.h>
 
+#include <vector>
+
 namespace filament::geometry {
 
 using namespace filament::math;
 using Algorithm = TangentSpaceMesh::Algorithm;
 
+template<typename T>
+class InternalArray {
+public:
+    void borrow(const T* ptr) noexcept {
+        mBorrowed = ptr;
+    }
+
+    T* allocate(size_t size) noexcept {
+        mAllocated.resize(size);
+        mAllocated.shrink_to_fit();
+        return mAllocated.data();
+    }
+
+    explicit operator bool() const noexcept {
+        return !mBorrowed && mAllocated.size() > 0;
+    }
+
+    const T* get() {
+        if (mBorrowed) {
+            return mBorrowed;
+        }
+        return mAllocated.data();
```

Comment:
Here you should assert() that the vector is not empty.

Reply:
done